### ISSUE TITLE
Crytical error of the plugin

### ISSUE THREAD

**Issue by MightCly** [diamond-rip/Eden#467]
Today I come to give a problem, which the plugin has, and it is the lack of basic placeholders.

I understand that currently the plugin is the most complete of all the free plugins out there so far, but honestly the issue of the few placeholders affects a lot, it doesn't matter if the plugin is very complete, if a user cannot modify his server to his style then he will stop using it and look for another alternative even if it is worse (In the case of some people I know)

By this I mean that, in the part of the messages that the plugin throws (that is, placeholders) it is a bit wrong, since there are no basic placeholders such as the one that gives if the player is ranked or unranked, that's good, it's , but to use it inside the plugin configuration, which prevents the user from adding custom things like placing it on a bossbar, tab or etc.

I suggest that for the next update add more basic and important placeholders, or even pass the ones that are already there, but that can be used only within the plugin to placeholders which can be placed in any other plugin that accepts placeholder api.

Here is a list of some basic ones in case nothing occurs to you, they may seem absurd or little used, but they are necessary:

- Countdown (The value set by the player)
- match type (Ranked/unranked)
- Party leader
- Queue type

At the moment I have these in mind, which I was unable to access since they are only usable to put them inside the plugin, or they are simply not there.

I also suggest some basic commands to add such as:

- /choosematch <mode> <kit> (This is for the party owner)

- /duel <player> <kit>

- /elo add <player> <amount>

- /elo remove <player> <amount>

Some things also seem absurd to me, with all due respect, such as the use of the uuid when accepting duels, as well as having to restart the server when creating a new arena, also having.

I'm just suggesting these things and I hope you don't take them the wrong way, some things ruin some issues to which you said you weren't going to add them, when the truth is they are necessary. I hope and they will be added in the next update, so as not to leave the plugin so bad in these aspects, because the truth is a good plugin, it just lacks a few things...

**Comment by Chromaah** [comment]
p.s there is already /duel...

**Comment by MightCly** [comment]
> p.s there is already /duel...

Yes but it opens the plugin menu, that is, eden's, I want to make my own, so I need to add the option to choose the arena from the same command example: /duel example builduhc

**Comment by liexes** [comment]
They are very good suggestions, but it is questioning that the owner @GoodestEnglish accepts them and adds

**Comment by MightCly** [comment]
I wait for responsive for @GoodestEnglish

**Comment by GoodestEnglish** [comment]
As the plugin updates, I will add more placeholder supports. 

I will also work on the elo command you've mentioned. Along with reset stats which mentioned in #378 

There's also a /choosematchtype command which allows party leader to choose match. You can view it in item.yml

**Comment by MightCly** [comment]
I know the commands are there, but there are no sub commands, for example the duel is /duel example and you get the menu with the modalities, what I want to be implemented is that you can send a duel and play in a party using commands, example:

- /duel example builduhc

-/choosematchtype split builduhc

I hope and this along with the placeholders mentioned by me will be added in the next update.

Thanks for the great plugin you developed.

@GoodestEnglish

**Comment by Chromaah** [comment]
yes that'd be cool for custom menus

**Comment by liexes** [comment]
+1
Yes, for custom menus

**Comment by MightCly** [comment]
> Today I come to give a problem, which the plugin has, and it is the lack of basic placeholders.
> 
> I understand that currently the plugin is the most complete of all the free plugins out there so far, but honestly the issue of the few placeholders affects a lot, it doesn't matter if the plugin is very complete, if a user cannot modify his server to his style then he will stop using it and look for another alternative even if it is worse (In the case of some people I know)
> 
> By this I mean that, in the part of the messages that the plugin throws (that is, placeholders) it is a bit wrong, since there are no basic placeholders such as the one that gives if the player is ranked or unranked, that's good, it's , but to use it inside the plugin configuration, which prevents the user from adding custom things like placing it on a bossbar, tab or etc.
> 
> I suggest that for the next update add more basic and important placeholders, or even pass the ones that are already there, but that can be used only within the plugin to placeholders which can be placed in any other plugin that accepts placeholder api.
> 
> Here is a list of some basic ones in case nothing occurs to you, they may seem absurd or little used, but they are necessary:
> 
> * Countdown (The value set by the player)
> * match type (Ranked/unranked)
> * Party leader
> * Queue type
> 
> At the moment I have these in mind, which I was unable to access since they are only usable to put them inside the plugin, or they are simply not there.
> 
> I also suggest some basic commands to add such as:
> 
> * /choosematch   (This is for the party owner)
> * /duel
> * /elo add
> * /elo remove
> 
> Some things also seem absurd to me, with all due respect, such as the use of the uuid when accepting duels, as well as having to restart the server when creating a new arena, also having.
> 
> I'm just suggesting these things and I hope you don't take them the wrong way, some things ruin some issues to which you said you weren't going to add them, when the truth is they are necessary. I hope and they will be added in the next update, so as not to leave the plugin so bad in these aspects, because the truth is a good plugin, it just lacks a few things...

Please, you can add the placeholders? Need to customize a thing in my server using actionbar.

**Comment by GoodestEnglish** [comment]
> > Today I come to give a problem, which the plugin has, and it is the lack of basic placeholders.
> > I understand that currently the plugin is the most complete of all the free plugins out there so far, but honestly the issue of the few placeholders affects a lot, it doesn't matter if the plugin is very complete, if a user cannot modify his server to his style then he will stop using it and look for another alternative even if it is worse (In the case of some people I know)
> > By this I mean that, in the part of the messages that the plugin throws (that is, placeholders) it is a bit wrong, since there are no basic placeholders such as the one that gives if the player is ranked or unranked, that's good, it's , but to use it inside the plugin configuration, which prevents the user from adding custom things like placing it on a bossbar, tab or etc.
> > I suggest that for the next update add more basic and important placeholders, or even pass the ones that are already there, but that can be used only within the plugin to placeholders which can be placed in any other plugin that accepts placeholder api.
> > Here is a list of some basic ones in case nothing occurs to you, they may seem absurd or little used, but they are necessary:
> > 
> > * Countdown (The value set by the player)
> > * match type (Ranked/unranked)
> > * Party leader
> > * Queue type
> > 
> > At the moment I have these in mind, which I was unable to access since they are only usable to put them inside the plugin, or they are simply not there.
> > I also suggest some basic commands to add such as:
> > 
> > * /choosematch   (This is for the party owner)
> > * /duel
> > * /elo add
> > * /elo remove
> > 
> > Some things also seem absurd to me, with all due respect, such as the use of the uuid when accepting duels, as well as having to restart the server when creating a new arena, also having.
> > I'm just suggesting these things and I hope you don't take them the wrong way, some things ruin some issues to which you said you weren't going to add them, when the truth is they are necessary. I hope and they will be added in the next update, so as not to leave the plugin so bad in these aspects, because the truth is a good plugin, it just lacks a few things...
> 
> Please, you can add the placeholders? Need to customize a thing in my server using actionbar.

What placeholder do you want? (Except countdown)

**Comment by MightCly** [comment]
> > > Today I come to give a problem, which the plugin has, and it is the lack of basic placeholders.
> > > I understand that currently the plugin is the most complete of all the free plugins out there so far, but honestly the issue of the few placeholders affects a lot, it doesn't matter if the plugin is very complete, if a user cannot modify his server to his style then he will stop using it and look for another alternative even if it is worse (In the case of some people I know)
> > > By this I mean that, in the part of the messages that the plugin throws (that is, placeholders) it is a bit wrong, since there are no basic placeholders such as the one that gives if the player is ranked or unranked, that's good, it's , but to use it inside the plugin configuration, which prevents the user from adding custom things like placing it on a bossbar, tab or etc.
> > > I suggest that for the next update add more basic and important placeholders, or even pass the ones that are already there, but that can be used only within the plugin to placeholders which can be placed in any other plugin that accepts placeholder api.
> > > Here is a list of some basic ones in case nothing occurs to you, they may seem absurd or little used, but they are necessary:
> > > 
> > > * Countdown (The value set by the player)
> > > * match type (Ranked/unranked)
> > > * Party leader
> > > * Queue type
> > > 
> > > At the moment I have these in mind, which I was unable to access since they are only usable to put them inside the plugin, or they are simply not there.
> > > I also suggest some basic commands to add such as:
> > > 
> > > * /choosematch   (This is for the party owner)
> > > * /duel
> > > * /elo add
> > > * /elo remove
> > > 
> > > Some things also seem absurd to me, with all due respect, such as the use of the uuid when accepting duels, as well as having to restart the server when creating a new arena, also having.
> > > I'm just suggesting these things and I hope you don't take them the wrong way, some things ruin some issues to which you said you weren't going to add them, when the truth is they are necessary. I hope and they will be added in the next update, so as not to leave the plugin so bad in these aspects, because the truth is a good plugin, it just lacks a few things...
> > 
> > 
> > Please, you can add the placeholders? Need to customize a thing in my server using actionbar.
> 
> What placeholder do you want? (Except countdown)

That was exactly what I needed, I was going to put it in the action bar of my server when starting a game. Please add it, many practice plugins have it by default.

But I would like you to add that one and these too:

%eden_match_spawn% to give the location of the spawn start of the player. example: -433 22 300
You could also do it separately, which would be a better example:

%eden_match_spawn_x% y and z

And also something that helps me a lot is if you add a command to finish the game, for example, the other day I was playing with my friends and I joined a mode that I didn't want, so I had to wait for them to finish to put another one. It would be good to add a command that ends the game, that can only be used by the party owner, example:

/party finish
